crowbar_framework/app/models/service_object.rb

```diff
@@ -989,20 +993,28 @@ def apply_role(role, inst, in_queue, bootstrap = false)
       end
     end
 
+    pre_cached_nodes = {}
+    filtered_elements = nil
+
     # When bootstrapping, we don't run chef, so there's no need for queuing
     if bootstrap
-      pre_cached_nodes = {}
       # do not try to process the queue in any case
       in_queue = true
     else
       # Attempt to queue the proposal.  If delay is empty, then run it.
       deps = proposal_dependencies(role)
+      filtered_elements = new_elements.deep_dup
```

Comment:
I tend to understand "filtered elements" as "elements that didn't pass the filter", and you use it for the opposite thing. Is it just me? :-)

Reply:
it may be, I changed it to cleaned elements